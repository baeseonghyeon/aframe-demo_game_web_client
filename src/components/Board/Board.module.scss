.container {
  background-color: #fff9ec;
  box-shadow: 0px 15px 0px 0px #ffb7aa;
  border-radius: 20px;
  width: 80%;
  // width: 100%;
  // min-width: 500px;
  max-width: 700px;
  max-height: 70vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding: 20px;
  position: relative;
  z-index: 20;

  .title {
    font-size: 2rem;
    color: white;
    text-shadow: 1px 3px 4px #da1f40;
    &__container {
      z-index: 1;
      position: absolute;
      top: -30px;
      background-color: #ff6380;
      padding: 0px 20px;
      border-radius: 20px;
      box-shadow: 0px 8px 0px 0px #ec002c;
      border-top: 3px solid #f64465;
    }
  }

  .wrapper {
    border: 5px solid #ffb7aa;
    border-radius: 20px;
    height: 100%;
    width: 100%;
    padding-top: 50px;
    padding-bottom: 30px;
    overflow: scroll;
    -ms-overflow-style: none; /* IE and Edge */
    scrollbar-width: none; /* Firefox */
    &::-webkit-scrollbar {
      display: none; /* Chrome, Safari, Opera*/
    }
  }
}
