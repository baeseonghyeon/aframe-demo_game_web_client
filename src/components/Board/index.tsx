import React, { Children } from 'react';
import styles from './Board.module.scss';
import cb from 'classnames/bind';

const cn = cb.bind(styles);

interface BoardProps {
  children?: React.ReactNode;
  title?: string;
  className?: string;
}

const Board = (props: BoardProps) => {
  const { children, title = '', className } = props;

  return (
    <div className={cn('container', className)}>
      {title && (
        <div className={cn('title__container')}>
          <h1 className={cn('title')}>{title}</h1>
        </div>
      )}
      <div className={cn('wrapper')}>{children}</div>
    </div>
  );
};

export default Board;
