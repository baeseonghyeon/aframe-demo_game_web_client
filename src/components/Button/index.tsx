import React from 'react';
import styles from './Button.module.scss';
import cb from 'classnames/bind';

const cn = cb.bind(styles);

interface ButtonProps {
  children?: React.ReactNode;
  className?: string;
  /**
   * 버튼 테마
   */
  theme?: 'pink' | 'yellow';
  /**
   * 버튼 곡선
   */
  shape?: 'circle';
}

const Button = (props: ButtonProps) => {
  const { theme = 'pink', shape, children, className } = props;

  return (
    <button
      className={cn(
        'container',
        `theme--${theme}`,
        `shape--${shape}`,
        className
      )}
    >
      {children}
    </button>
  );
};

export default Button;
