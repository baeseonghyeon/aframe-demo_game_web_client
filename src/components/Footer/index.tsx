import React from 'react';
import styles from './Footer.module.scss';
import cb from 'classnames/bind';

const cn = cb.bind(styles);

const Footer = () => {
  return <div className={cn('container')}>Â© 2021 A-Frame Demo Game</div>;
};

export default Footer;
