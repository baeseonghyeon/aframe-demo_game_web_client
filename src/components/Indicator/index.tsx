import React from 'react';
import styles from './Indicator.module.scss';
import cb from 'classnames/bind';

const cn = cb.bind(styles);

const Indicator = () => {
  return (
    <svg className={cn('indicator')} viewBox="0 0 100 100">
      <circle className={cn('circle')} cx="50" cy="50" r="45"></circle>
    </svg>
  );
};

export default Indicator;
