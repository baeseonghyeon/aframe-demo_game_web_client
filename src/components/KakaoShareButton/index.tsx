import React, { useEffect } from 'react';
import styles from './KakaoShareButton.module.scss';
import cb from 'classnames/bind';
import kakaoIcon from './assets/kakao.png';

const cn = cb.bind(styles);

interface KakaoShareButtonProps {
  children?: React.ReactNode;
  className?: string;
  onClick?: () => void;
}

const KakaoShareButton = (props: KakaoShareButtonProps) => {
  const { children, className, onClick } = props;

  useEffect(() => {
    createKakaoButton();
  }, []);

  const createKakaoButton = () => {
    if (window.Kakao) {
      const kakao = window.Kakao;
      // 중복 initialization 방지
      if (!kakao.isInitialized()) {
        kakao.init(process.env.REACT_APP_KAKAO_KEY);
      }

      kakao.Link.createCustomButton({
        container: '#kakao-link-btn',
        templateId: 56200
      });
    }
  };

  return (
    <div className={cn('container')}>
      {/* Kakao share button */}
      <button id="kakao-link-btn" className={cn('btn')}>
        <img className={cn('icon')} src={kakaoIcon} alt="kakao-share-icon" />
      </button>
    </div>
  );
};

export default KakaoShareButton;
