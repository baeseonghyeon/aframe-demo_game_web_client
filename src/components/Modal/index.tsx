import React from 'react';
import styles from './Modal.module.scss';
import cb from 'classnames/bind';

const cn = cb.bind(styles);

export interface ModalProps extends React.HtmlHTMLAttributes<HTMLDivElement> {
  /**
   * 컨테이너 클래스 네임
   */
  containerClassName?: string;
  /**
   * 모달 여부
   */
  isVisible: boolean;
  /**
   * 백드롭 여부
   */
  isBackdrop?: boolean;
  /**
   * 백드롭 스타일
   */
  backdropClassName?: string;
  /**
   * 백드롭 클릭 이벤트
   */
  onClickBackdrop?: () => void;
  /**
   * ref
   */
  ref?: any;
}

/**
 *  모달 컴포넌트 props : ({ isVisible, onClickBackdrop, backdropClassName })
 */
const Modal = (props: ModalProps) => {
  const {
    className,
    containerClassName,
    isVisible = false,
    isBackdrop = true,
    backdropClassName,
    onClickBackdrop,
    ref,
    ...rest
  } = props;

  // Visible Class
  const visibleClassNames = cn([
    containerClassName && containerClassName,
    isVisible ? `container--visible` : `container--none`,
    isVisible && isBackdrop ? `container--fill` : `container--fit`
  ]);

  // Modal Container Class
  const classNames = cn(props.className, 'wrapper');

  return (
    <div className={visibleClassNames} ref={ref} {...rest}>
      {/* Modal Content  */}
      <div className={classNames}>{isVisible && props.children}</div>
      {/* BackDrop Layout */}
      {isBackdrop && (
        <div
          className={cn('backdrop', backdropClassName)}
          onClick={onClickBackdrop}
        ></div>
      )}
    </div>
  );
};

export default Modal;
