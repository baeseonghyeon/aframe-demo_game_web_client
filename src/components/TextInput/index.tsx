import React from 'react';
import styles from './TextInput.module.scss';
import cb from 'classnames/bind';

const cn = cb.bind(styles);

export interface TextInputProps
  extends React.HtmlHTMLAttributes<HTMLDivElement | HTMLInputElement> {
  className?: string;
  placeholder?: string;
  value: string;
  autoFocus?: boolean;
  onChange?: (e: any) => void;
}

const TextInput = (props: TextInputProps) => {
  const { placeholder, value, className, autoFocus, onChange, ...rest } = props;

  return (
    <input
      className={cn('container', className)}
      placeholder={placeholder}
      value={value}
      onChange={onChange}
      autoFocus={autoFocus}
      {...rest}
    />
  );
};

export default TextInput;
