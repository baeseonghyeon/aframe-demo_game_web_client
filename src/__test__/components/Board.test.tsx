import React from 'react';
import ReactDOM from 'react-dom';
import { default as Board } from '../../components/Board';
import { render } from '@testing-library/react';

describe('[Component] Board', () => {
  const __TEST_BOARD_TITLE = '__test__';
  it('스냅샷 테스트', () => {
    const rendered = render(
      <Board title={__TEST_BOARD_TITLE}>
        <p>BOARD COMPONENT</p>
      </Board>
    );
    expect(rendered.container).toMatchSnapshot();
  });

  it('프로퍼티 테스트 : title', () => {
    const { getByText } = render(<Board title={__TEST_BOARD_TITLE} />);
    getByText(__TEST_BOARD_TITLE);
  });

  it('프로퍼티 테스트 : children', () => {
    const CHILDREN_TEXT = `${__TEST_BOARD_TITLE}__BOARD_CHILDREN__${new Date().getTime()}`;
    const { getByText } = render(
      <Board>
        <p>{CHILDREN_TEXT}</p>
      </Board>
    );
    getByText(CHILDREN_TEXT);
  });
});
