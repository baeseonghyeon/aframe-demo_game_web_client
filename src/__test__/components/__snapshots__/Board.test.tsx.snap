// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] Board 스냅샷 테스트 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="title__container"
    >
      <h1
        class="title"
      >
        __test__
      </h1>
    </div>
    <div
      class="wrapper"
    >
      <p>
        BOARD COMPONENT
      </p>
    </div>
  </div>
</div>
`;
